src/Microsoft.Identity.Client/AccountId.cs

```diff
@@ -70,6 +70,15 @@ public AccountId(string identifier, string objectId, string tenantId)
             ValidateId();
         }
 
+        /// <summary>
+        /// Constructor of an AccountId meant for Adfs scenarios since Adfs instances lack tenant ids.
+        /// </summary>
+        /// <param name="identifier">Unique identifier for the account</param>
+        public AccountId(string adfsIdentifier)
```

Comment:
>adfsIdentifier [](start = 32, length = 14)

rename: identifier. make sure the summary/remarks identifies this is for ADFS. Alternatively create a factory method called CreateAccountIdForADFS

Reply:
I suggest you make this private and make a factory method instead that can create an AdfsAccountId, e.g. CreateAdfsAccountId(identifier)

---
In reply to: [250830377](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r250830377) [](ancestors = 250830377)

Reply:
Can you make this private and create a factory method instead as suggested here? this will make it more clear that this is meant only to be used in the ADFS case.

---
In reply to: [252131037](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r252131037) [](ancestors = 252131037,250830377)